src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java

```diff
@@ -75,9 +78,17 @@ public void configureCoreProperties(
   private void configureFactories() {
     project.afterEvaluate(
         project -> {
-          // create the sdk builder factory after we know the location of the sdk
-          cloudSdkBuilderFactory = new CloudSdkBuilderFactory(toolsExtension.getCloudSdkHome());
-          managedCloudSdkFactory = new ManagedCloudSdkFactory(toolsExtension.getCloudSdkVersion());
+          try {
+            managedCloudSdk =
+                new ManagedCloudSdkFactory(toolsExtension.getCloudSdkVersion()).newManagedSdk();
+            cloudSdkBuilderFactory =
+                new CloudSdkBuilderFactory(
+                    toolsExtension.getCloudSdkHome() == null
+                        ? managedCloudSdk.getSdkHome().toFile()
+                        : toolsExtension.getCloudSdkHome());
+          } catch (UnsupportedOsException | BadCloudSdkVersionException ex) {
```

Comment:
I wonder if we can just set the managedCloudSdk conditionally here if `getCloudSdkHome == null`

Reply:
We potentially don't want to be throwing a BadCloudSdkVersionException if the user isn't using the managed Cloud SDK.